### ISSUE TITLE
[a11y]: DataTable [all variants with search & toolbar with overflow] - SR reading visually hidden element

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#14274]
### Package

@carbon/react

### Browser

Safari & iPhone

### Package version

@carbon/react@1.32.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO

### Description
Settings menu in the toolbar region
Options menu in the last column of the table

VO identifies "Focus sentinel" link at the beginning and ending of the dropdown

After expanding the mentioned dropdown, when we navigate using Left/Right swipe, VO announces "Focus sentinel" link at the beginning and ending of the dropdown. Along with that, The first option is announced as "selected" when we swipe Right from the "focus sentinel" link present at the beginning of the dropdown. similarly, The last option is announced as "selected" when we swipe Left from the "focus sentinel" link present at the end of the dropdown. Note: This issue occurs on the "settings" menu present in all the variants.

### Recommended Fix
1. SR is not expected to identify the visually hidden element
2. If the selected status is programmatically associated, it should be in accordance with the element selection.

If the screen reader identifies the visually hidden content, it tends to mislead users who are blind. 

### WCAG 2.1 Violation
4.2.1 Name, Role, and Value

### Reproduction/example
Record `Number 80` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

- Toolbar with overflow: https://react.carbondesignsystem.com/?path=/story/components-datatable-toolbar--with-overflow-menu
- Search - filtering: https://react.carbondesignsystem.com/?path=/story/components-datatable-filtering--default
- Search - dynamic: https://react.carbondesignsystem.com/?path=/story/components-datatable-dynamic--default
- Search - batch actions: https://react.carbondesignsystem.com/?path=/story/components-datatable-batch-actions--default

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
